app/src/main/java/org/fossasia/pslab/activity/WaveGeneratorActivity.java

```diff
@@ -471,20 +528,24 @@ public void selectBtn(WaveConst btn_selected) {
                 pwmBtnActive = WaveConst.SQ2;
                 activePwmPinTv = pwmMonSqr2;
                 activePwmPinTv.setEnabled(true);
-                pwmMonSelectPin.setText(getString(R.string.text_sq2));
-                pwmBtnPhase.setEnabled(true);
 
-                value = (double) WaveGeneratorCommon.wave.get(pwmBtnActive).get(WaveConst.FREQUENCY);
-                valueText = value + " " + getString(R.string.unit_hz);
-                pwmFreqValue.setText(valueText);
+                if (digital_mode == WaveConst.SQUARE) {
+                    value = (double) WaveGeneratorCommon.wave.get(pwmBtnActive).get(WaveConst.FREQUENCY);
+                    valueText = value + " " + getString(R.string.unit_hz);
+                    pwmFreqValue.setText(valueText);
+
```

Comment:
Additional line. Also as I've mentioned in Gitter channel, this if block and the block starting from `543` does almost the same thing. You could have exported them to a method and call with relevant parameters.

Reply:
@CloudyPadmal I will refractor the complete code of the wave generator in one PR, I will open an issue for that.  I have done the other changes.

Reply:
and for the future PRs I will keep in mind your comment on the gitter channel.

Reply:
I think you can just run a simple refactor in this PR itself. No need of a new PR to remove extra lines and spaces.